### ISSUE TITLE
onnx ensemble 

### ISSUE THREAD

**Issue by PlakiaM** [mammoth-eu/mammoth-commons#90]
Hello all, 
i am getting the following error will i am testing onnx ensemble 

![Image](https://github.com/user-attachments/assets/a85033b9-d5ba-435e-a2ee-851886c5ba72)
i believe that this is an issue of the type of algorithm we trained, logistic regression which does not have alpha estimators which seems to be a requirement from my understanding.
Could this issue to be addressed?
Thank you!

**Comment by arjunroyihrpa** [comment]
@PlakiaM this is my account. You can tag me with this id.

**Comment by arjunroyihrpa** [comment]
@PlakiaM I made the changes where now you will be able to use the onnx_ensemble predictor with your local trained any sklearn model. For data loading when working locally, please use the data_read_any loader from the data_any module in order to not have clean preprocessed data. If you still find the error, no worries, I will update the notebook instruction shortly (by next week) to help users understand how to train any model, deploy in onnx, and use the modules.

**Comment by maniospas** [comment]
Note that I need to create a new release before the update is accessible to upgrade via pi. This will happen by Monday because I am working on several issues that affect each other simultaneously.

**Comment by maniospas** [comment]
I released new major versions of everything. You can now test @arjunroyihrpa 's changes @PlakiaM . Do resolve the issue if things are working.

Future updates will happen much more frequently - current ones had exceptional circumstances, including waiting for an issue to be fixed by a third-party library. (Unrelated: UI will be further improved for the demonstrator's dashboard but I did not focus on that till now.)

Unfortunately, sweeping changes to the Python ecosystem mean that you need to work with Python 3.13 now to install everything and have it work properly. To prevent a lot of issues I made this a hard requirement for the mai-bias package. Good news are that modules install the packages as needed, so you will have less bloat than previously if you don't run the vision modules.

**Comment by PlakiaM** [comment]
OK, thank you both for letting me know. A question here, do ew have any restrictions about the fairbench version? or should i use the most updated?

**Comment by maniospas** [comment]
Hi again! 

Finally resolved the issue of dependencies in a manner that does not wait for the next release of third-party libraries if needed (the demonstrator now installs some corresponding dependencies from github repos where issues are fixed when running corresponding modules).

It's generally a good idea to use the latest version of fairbench, as I keep adding modules based on its functionality too.

Created a release with everything (that is, all tests) working.

**Comment by arjunroyihrpa** [comment]
<img width="1507" alt="Image" src="https://github.com/user-attachments/assets/854bb842-32dc-45ea-ba99-422a13b8280f" />

This is with mmm_Fair_GBT. I have also uploaded a notebook to guide users how to train and upload model locally (I used a Logistic Regression model, you can also use a tensorflow or a pytorch model, with appropriate onnx converter). You can follow that notebook to train and test your model on your local data.

**Comment by PlakiaM** [comment]
Hello all, 
i am running the pipeline ONNX ensemble + multi objective report, but there are no visualizations. 

<img width="443" height="300" alt="Image" src="https://github.com/user-attachments/assets/d9d42eaa-8ea5-4eb9-a02e-8a086d051e53" />
The only logs that i have are the following:
(py313) C:\Users\m.plakia>python -m mai_bias.app
[23316:17268:0915/132857.325:ERROR:gpu_channel_manager.cc(956)] Failed to create GLES3 context, fallback to GLES2.
[23316:17268:0915/132857.326:ERROR:gpu_channel_manager.cc(967)] ContextResult::kFatalFailure: Failed to create shared context for virtualization.

I am not quite sure if they help.
Thank you all!

**Comment by arjunroyihrpa** [comment]
Dear @PlakiaM,

As I am currently employed at a different company, during weekdays I will not get time to work on it. However, on the weekend I will work on this and try to resolve the issue. 

Best,
Arjun

**Comment by arjunroyihrpa** [comment]
Hi @PlakiaM,

Apologies for the delay in addressing this. The issue has been identified, and resolving it will require an app-level decision from @maniospas.

The problem is not with the component itself, it works as intended. The issue arises because the app’s UI is unable to render Plotly-generated 3D plots correctly. One factor is the include_plotlyjs setting, which can be handled in the metric component by using:
`include_plotlyjs="cdn"`
instead of the current` include_plotlyjs=True`.

However, the main limitation is that the app currently does not support WebGL, which is required for 3D plot rendering. This is caused by the following environment settings in the app:

```
os.environ["QT_QUICK_BACKEND"] = "software"
os.environ["QTWEBENGINE_CHROMIUM_FLAGS"] = "--disable-gpu"
```

These settings prevent WebGL from functioning. Since this is an app-level restriction, it needs to be addressed or decided by the maintainer.

Recommended steps to fix:

1. Comment out the two environment variable settings above.
2. Update the metric component to use `include_plotlyjs="cdn" `
(I will do this by pushing the update in mmm_fair_cli)

Optionally, I can adjust some CSS in the metric component to improve appearance.

Please check the attached screenshots for reference.

<img width="1690" height="928" alt="Image" src="https://github.com/user-attachments/assets/ecd47dea-9f30-46a5-9eb1-4316100f23cb" />
<img width="1719" height="1060" alt="Image" src="https://github.com/user-attachments/assets/ae8c26ff-5000-408f-a7a4-f90ba972844c" />

<img width="668" height="151" alt="Image" src="https://github.com/user-attachments/assets/13df76e0-bbb1-478d-af44-0e928b0ba7ca" />

.

**Comment by arjunroyihrpa** [comment]
Hi @maniospas,

I also want to point out that whoever administers or maintains the software/app should consider version sets and regression testing before making any global updates.

Regression tests should ideally be covered in our integration testing. However, in this case, we see that some global changes affected my component: it worked when I initially pushed it, but stopped working after global updates. This should have been caught during integration testing, the component itself passes tests, but rendering in the app environment fails.

I understand it can be difficult to foresee every angle that needs coverage, but a version set of the previously working state should always be available to help track and prevent such issues.

**Comment by maniospas** [comment]
Hi, @arjunroyihrpa !

The original expectation was that there would be ongoing maintenance and support by each module's creator as the project goes on because we are following an iterative development scheme. The particular decision is in there due to issues that some platforms have with webgl rendering (which prevented certain partners from running the demonstrator at all).

I must stress that, with our resources, it is completely unrealistic to expect a full round of testing to be concluded before having a release. In fact, it takes so much time to test everything, that we have barely finished the *second* round of tests, corresponding only to major releases of the main toolkit. We have had many more minor releases in the interim, some of which fixed very critical issues in the interfaces and working mechanism. Do note that we have so far fixed 52 closed issues (and a comparable number of informal ones) and maybe only 1 or 2 of those -though I donnot remember exact details- were due to breaking changes from new releases (I do not count cases where modules polluted the interfaces of others, so the quick fix consisted of reverting the pollution and requiring debugging the offendant module).

Finally, we have a very simple mechanism for rolling back to a stable version state if you want to: switching to a known working commit version. One can also install a specific version of the demonstrator with pip (https://pypi.org/project/MAI-Bias/#history ). 

Lastly: I guess we can have two versions for starting the UI: `python -m mai_bias.app`, `python -m mai_bias.app_safe` (with webgl disabled) and in the latter I can add a button so that demonstrator users can directly open results in their browser for similar issues.

I will send an update here once I've done this.

**Comment by maniospas** [comment]
Released the version of commons I had been working with a bit pre-emptively to allow testing for the above. In my machine, it does not matter if there is webgl enabled or disabled - pyside just cannot render it. That said, I have added a button from which one can open the result in the browser, and there it renders properly.

**Comment by maniospas** [comment]
Closing this issue because in my view it has been addressed = through bilateral call the methodology was not applicable and the final rendering only had one point.
